### PR TITLE
virtio_fs: compare symbolic link info in host and Windows guest

### PR DESCRIPTION
**Pull request by xiagao**
Compare symbolic link information in the Windows host and guest to check if they are consistent.
ID: 1690

### CODE REVIEW

**Comment by xiagao** [comment]
Test pass on x86_64 with windows guest and linux guest.

**Comment by xiagao** [comment]
@zhenyzha Could you help to check linux guest part, thanks.

File: qemu/tests/virtio_fs_share_data.py [hunk 1]
```diff
@@ -555,6 +558,36 @@ def __file_check(file, guest=None):
                                       "host and guest are inconsistent")
                         session.cmd("cd -")
                     else:
+                        def __file_check(sym_link, guest=None):
```
Comment (zhenyzha) [inline qemu/tests/virtio_fs_share_data.py:561]: @xiagao I think we can make def __file_check() support both Linux and win
Simple example:
```
        def __file_check(file, os_type, guest=None):
            cmd_map = {'win_host':'cat %s',
                           'win_guest':'type %s',
                           'linux_host' ='ls -l %s',
                           'linux_guest' ='ls -l %s'}
            
            cmd = '_'.join((os_type, guest))
            if guest:
                o = session.cmd_output(cmd_map[cmd] % file)
            else:
                o = process.run(cmd_map[cmd] % file).stdout_text
            return o.strip().split()[-1]
```

@YongxueHong What do you think?

Comment (YongxueHong) [reply qemu/tests/virtio_fs_share_data.py:561]: Hi @zhenyzha 
yeah, it's okay for me if we are gonging to support the Linux guest as well.

Comment (xiagao) [reply qemu/tests/virtio_fs_share_data.py]: @zhenyzha Updated, could you review it again？

**Comment by zhenyzha** [comment]
Pass on the arm platform

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Review by zhenyzha — APPROVED** [review]

**Review by YongxueHong — APPROVED** [review]

**Comment by YongxueHong** [comment]
Hi @fbq815 
Would you like to help review it?
Thanks.

**Comment by fbq815** [comment]
Tests PASS on s390x for both RHEL8 and RHEL9

**Pull request merged**